### PR TITLE
[BugFix] `get_json_string` parse json field in non-string type as string

### LINKED ISSUE #6447: `get_json_string` cannot parse json field in non-string type
### Steps to reproduce the behavior (Required)
```
mysql> select get_json_string('{"key1":{"key2":       1}}', '$.key1');

```

### Expected behavior (Required)
```
+-------------------------------------------------------------------+
| get_json_string('{"key1":{"key2":                 1}}', '$.key1') |
+-------------------------------------------------------------------+
| {"key2":1}                                                        |
+-------------------------------------------------------------------+
```
### Real behavior (Required)
```
+-------------------------------------------------------------------+
| get_json_string('{"key1":{"key2":                 1}}', '$.key1') |
+-------------------------------------------------------------------+
| NULL                                                              |
+-------------------------------------------------------------------+
```
### StarRocks version (Required)
main, branch-2.2, branch-2.1

### PR DESCRIPTION
**Pull request by rickif**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #6447,#6426

## Problem Summary(Required) ：

This makes `get_json_string` parse json field in non-string type as string.

### CODE REVIEW

File: be/test/exprs/vectorized/json_functions_test.cpp [hunk 1]
```diff
@@ -164,6 +164,44 @@ TEST_F(JsonFunctionsTest, get_json_stringTest) {
...
+TEST_F(JsonFunctionsTest, get_json_string_casting) {
+    std::unique_ptr<FunctionContext> ctx(FunctionContext::create_test_context());
+    Columns columns;
+    auto strings = BinaryColumn::create();
+    auto strings2 = BinaryColumn::create();
+
+    std::string values[] = {
+            "{\"k1\":    1}", //int
+            "{\"k1\":    3.14159}", //double
+            "{\"k1\":    {\"k2\":       \"v2\"}}"}; //object
```
Comment (stdpain) [inline be/test/exprs/vectorized/json_functions_test.cpp:176]: add a case: '{"key1": "{\"key2\":\"value2\"}"}'

File: be/src/exprs/vectorized/json_functions.cpp [hunk 2]
```diff
@@ -322,43 +322,63 @@ ColumnPtr JsonFunctions::_iterate_rows(FunctionContext* context, const Columns&
...
+    result.append(d);
+    return;
+}
 
-        result.append(d);
+template <>
+void JsonFunctions::_build_column(ColumnBuilder<TYPE_VARCHAR>& result, simdjson::ondemand::value& value) {
+    simdjson::ondemand::json_type tp;
+    auto err = value.type().get(tp);
+    if (UNLIKELY(err)) {
```
Comment (mofeiatwork) [inline be/src/exprs/vectorized/json_functions.cpp:355]: How about simplify it with a macro?

**Review by murphyatwork — DISMISSED** [review]

File: be/src/exprs/vectorized/json_functions.cpp (hunk 2 above)
Comment (rickif) [reply be/src/exprs/vectorized/json_functions.cpp]: Great idea. I have replaced it with a macro `APPEND_NULL_AND_RETURN_IF_ERROR`

**Comment by rickif** [comment]
run starrocks_clang-format

**Review by stdpain — APPROVED** [review]

**Review by murphyatwork — APPROVED** [review]

**Pull request merged**